### PR TITLE
My solution to airport challenge

### PR DESCRIPTION
**Pull request by DawidSzpener**
Sunday : 14:05

### CODE REVIEW

File: spec/airport_spec.rb [hunk 1]
```diff
@@ -33,24 +36,29 @@
...
+      expect { london.land(Plane.new, stormy) }.to raise_error "Stormy weather! Cant land"
+    end
+    
     it "should let flying planes to land at airport" do
       london = Airport.new
       london.create_plane(plane = Plane.new)
-      london.takeoff(plane)
-      expect(london.land(plane)).to eq plane
+      london.takeoff(plane, sunny)
+      expect(london.land(plane, sunny)).to eq plane
```
Comment (Jamie95187) [inline spec/airport_spec.rb]: The error could maybe expect to return an increment to the size of the instance variable @planes in the airport object :)

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -33,24 +36,29 @@
   
   describe 'land' do
     
+    it "should not allow plane to land when weather is stormy" do
+      london = Airport.new
+      expect { london.land(Plane.new, stormy) }.to raise_error "Stormy weather! Cant land"
+    end
+
```
Comment (Jamie95187) [inline spec/airport_spec.rb]: Could consider using context block to group when testing a given scenario (such as weather = stormy) 👍

File: lib/airport.rb [hunk 3]
```diff
@@ -17,15 +18,17 @@ def create_plane(plane)
     return @plane
   end
   
-  def land(plane)
+  def land(plane, weather)
+    raise("Stormy weather! Cant land") if weather.stormy? == true
     raise("Airport is full") if @planes.size >= @capacity
     raise("Plane is already at airport") unless $flying_planes.include?(plane)
     
     @planes << $flying_planes.slice($flying_planes.index(plane))
```
Comment (Jamie95187) [inline lib/airport.rb]: Consider SRP, could create a new method for example update_planes_at_airport :)

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    $flying_planes.delete_at($flying_planes.index(plane))
+  end
+  
+  def takeoff(plane, weather)
+    raise("Stormy weather! Cant take off now") if weather.stormy? == true
+    raise("This plane is already flying") if $flying_planes.include?(plane)
+    raise("No such plane in this airport") unless @planes.include?(plane)
+    
+    $flying_planes << @planes.slice(@planes.index(plane))
+    @planes.delete_at(@planes.index(plane))
```
Comment (alicelieutier) [inline lib/airport.rb]:
```rb
$flying_planes << plane
@planes.delete plane
```

**Pull request closed without merging**